kubernetes/src/main/java/io/kubernetes/client/openapi/JSON.java

```diff
@@ -42,11 +44,22 @@
 
   private boolean isLenientOnJson = false;
 
+  public static final DateTimeFormatter RFC3339MICRO_FORMATTER =
+      new DateTimeFormatterBuilder()
+          .parseDefaulting(ChronoField.OFFSET_SECONDS, 0)
+          .append(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss"))
+          .optionalStart()
+          .appendFraction(ChronoField.NANO_OF_SECOND, 6, 6, true)
+          .optionalEnd()
+          .appendLiteral("Z")
+          .toFormatter();
+
```

Comment:
I'm not sure this is necessary. Why do we need an explicit formatter for a format that already works by default with `OffsetDateTime` (I believe)?

Reply:
kubernetes server-side requires 6-digit fraction for micro timestamp types (i.e. LeaseSpec#accquireTime) where "0" should be padded in the end if necessary. however, the default `DateTimeFormatter.ISO_OFFSET_DATE_TIME` doesn't support right-padding for micro-sec timestamps. e.g. by the default formatter`2018-04-03T11:32:26.1234Z` is a valid timestamp, but kubernetes server will be rejecting b/c lacking right-padded "0"s, it's expected to be `2018-04-03T11:32:26.123400Z`